packages/editor/src/components/post-actions/index.js

```diff
@@ -28,20 +32,28 @@ const {
 
 export default function PostActions( { onActionPerformed, buttonProps } ) {
 	const [ isActionsMenuOpen, setIsActionsMenuOpen ] = useState( false );
-	const { item, postType } = useSelect( ( select ) => {
+	const { item, postType, resource } = useSelect( ( select ) => {
 		const { getCurrentPostType, getCurrentPost } = select( editorStore );
+		const { getPostType } = select( coreStore );
+		const _postType = getCurrentPostType();
+		const _resource = getPostType( _postType )?.rest_base;
+
 		return {
 			item: getCurrentPost(),
-			postType: getCurrentPostType(),
+			postType: _postType,
+			resource: _resource,
 		};
 	}, [] );
 	const allActions = usePostActions( postType, onActionPerformed );
+	const permissions = useResourcePermissions( resource, item.id );
```

Comment:
How important is the "item.id" here? My alternative solution would only work if we had the same permissions for all items.

Reply:
I think it's important for some roles that can delete only some items for example.

>My alternative solution would only work...

Do you have a PR?

I tried today a bit what you suggested about wrapping the action with a permissions check, but couldn't figure out how to handle this in Data Views - that means without checking each item specifically outside of `isEligible` function.

Reply:
I don't have a PR, it would be very similar to this one, just moving this call to with usePostActions. 

That said, yeah if we want per item checks, we'd have to follow something like the current PR but it would mean a lot of REST API calls unless we wait for the full core-data based solution and have the `isEligible` support thunks.

Reply:
Specific actions can be restricted only for specific entities, so the `item.id` check is important for `edit/update/delete` actions.